sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1088,26 +1112,9 @@ class Dataset[T] private[sql](
     // Otherwise, find the trivially true predicates and automatically resolves them to both sides.
     // By the time we get here, since we have already run analysis, all attributes should've been
     // resolved and become AttributeReference.
-    val resolver = sparkSession.sessionState.analyzer.resolver
```

Comment:
we don't look at the `dataFrameSelfJoinAutoResolveAmbiguity` config here. Why do we check the config in `joinWith`?